### PR TITLE
Samtools `coverage` submodule

### PR DESCRIPTION
**Pull request by vladsavelyev**
Parse `samtools coverage` output:

```
#rname  startpos  endpos  numreads  covbases  coverage  meandepth  meanbaseq  meanmapq
chr1           1   40001      5640      1181  2.95243   16.7743    40.9             60
chr2           1   40001      5640      1181  2.95243   16.7743    40.9             60
```

Adds:
* Line plot with regions along the x-axis, and metrics as a switch button panel,
* Table/violin summing/averaging across all regions, 
* "Mean Depth" in gen stats.

Screenshot 2024-02-20 at 14 41 35

Fixes https://github.com/MultiQC/MultiQC/issues/2021

### CODE REVIEW

**Comment by vladsavelyev** [comment]
@multiqc-bot changelog

**Comment by ewels** [comment]
Seems unrelated to this PR.. Get the same thing on `main`

**Review by ewels — APPROVED** [review]
LGTM! Would be good to clean up the regex escape thing, but otherwise good to merge.

File: multiqc/modules/samtools/rmdup.py [hunk 1]
```diff
@@ -68,14 +68,14 @@ def parse_samtools_rmdup(self):
         # General Stats Table
         stats_headers = {
             "pct_dups": {
-                "title": "% Dups",
+                "title": "Dups",
```
Comment (ewels) [inline multiqc/modules/samtools/rmdup.py:71]: Maybe we can use a few more characters here..
```suggestion
                "title": "Duplicates",
```

**Comment by vladsavelyev** [comment]
Fixed the warning 👍

**Pull request merged**